tests/whisper_protocol/crypto_wrappers/curve25519.rs

```diff
@@ -54,19 +59,74 @@ pub fn generate_private_key() -> PrivateKey{
     PrivateKey::from_bytes(*private_key)
 }
 
-#[derive(Clone)]
-#[derive(PartialEq)]
 pub struct PublicKey {
```

Comment:
The  issue I see is that to my knowledge we cannot access a packed struct as a [u8;33] adding complexity and copies. 

Given the key is always accessed as a 33 byte array literal except when its passed to the external curve library, storing it as anything else seems unnecessarily complicated to me. Thoughts?